server/controllers/cohort.js

```diff
@@ -50,8 +51,46 @@ const getCohortById = (req, resp) => {
   });
 };
 
+const updateCohortById = (req, resp) => {
+  const { description, name } = req.body;
+  const { id: cohortId } = req.params;
+
+  const dataToUpdate = filter(x => x !== undefined)({
+    description,
+    name
+  });
+
+  Cohort.findOneAndUpdate(
+    {
+      _id: cohortId,
+      $or: [{ adminIds: req.user._id }]
+    },
+    dataToUpdate,
+    { new: true },
+    (err, doc) => {
+      if (!doc) {
+        return resp.status(404).send({
+          message: 'You have no permissions to perform this action'
+        });
+      }
+
+      if (err) {
```

Comment:
move error checking to the top - in current case if we have error then we will send 404 from the first if